crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -267,31 +269,59 @@ impl From<SyncResponse> for enums::RefundStatus {
     }
 }
 
-impl<F, T> TryFrom<types::ResponseRouterData<F, DatatransResponse, T, types::PaymentsResponseData>>
-    for types::RouterData<F, T, types::PaymentsResponseData>
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            DatatransResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsAuthorizeData, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: types::ResponseRouterData<F, DatatransResponse, T, types::PaymentsResponseData>,
+        item: types::ResponseRouterData<
+            F,
+            DatatransResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
     ) -> Result<Self, Self::Error> {
-        let resource_id = match item.response.transaction_id.as_ref() {
-            Some(transaction_id) => {
-                types::ResponseId::ConnectorTransactionId(transaction_id.to_string())
+        let status = enums::AttemptStatus::foreign_from((
+            &item.response,
+            item.data.request.is_auto_capture()?,
+        ));
+        let response = match &item.response {
+            DatatransResponse::ErrorResponse(error) => {
+                let error_message = error.message.clone();
+                Err(types::ErrorResponse {
+                    code: error.code.clone(),
+                    message: error_message.clone(),
+                    reason: Some(error_message),
+                    attempt_status: None,
+                    connector_transaction_id: None,
+                    status_code: item.http_code,
+                })
+            }
+            DatatransResponse::TransactionResponse(response) => {
+                Ok(types::PaymentsResponseData::TransactionResponse {
+                    resource_id: types::ResponseId::ConnectorTransactionId(
+                        response.transaction_id.clone(),
+                    ),
+                    redirection_data: None,
+                    mandate_reference: None,
+                    connector_metadata: serde_json::to_value(response).ok(),
```

Comment:
Why are we doing this?

Reply:
Storing extra fields from the DataTrans Auth Success Response here
AcquirerAuthorizationCode